### PR TITLE
Use K8s 1.10.4 client

### PR DESCRIPTION
**Pull request by esnible**
Kubernetes 1.10 introduces _config.ExecProvider_, a mechanism to allow Kubernetes client config to specify a binary that is used on the client side to get an authentication token.

Resolves Issue https://github.com/istio/istio/issues/5327

No changes to Istio seem to be needed.  I don't have AWS or the ability to test this interactively but I verified following the instructions on the Issue and doing `istioctl --context <mycontext> --log_output_level debug get routerules` that using the newer K8s client code shows an exec failure for the Heptio client in the logs.  Previously _istioctl_ had not tried to exec anything.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -38,14 +38,13 @@ required = [
...
-
 # client-go release-6.0 corresponds to k8s release-1.9 branch. See
-# https://github.com/kubernetes/client-go/tree/release-6.0#compatibility-matrix
+# https://github.com/kubernetes/client-go#compatibility-matrix
 # for details.
 [[override]]
   name = "k8s.io/client-go"
-  branch = "release-6.0"
+#  branch = "release-6.0"
+  branch = "release-7.0"	# For K8s 1.10.x
```
Comment (ayj) [inline Gopkg.toml]: Can you remove the commented out line and comment reference to release-6.0?

**Comment by cgfulton** [comment]
@esnible Successfully ran `istioctl create -f samples/bookinfo/routing/bookinfo-gateway.yaml` using your running against in AWS EKS.  Before running the same command I got `Error: gateways.networking.istio.io is forbidden: User "system:anonymous" cannot create gateways.networking.istio.io in the namespace "default"`. I printed out my Kubernetes versions and heptio-authenticator-aws info for completeness. 

Kubernetes Info
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T15:22:13Z", GoVersion:"go1.9.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-28T20:13:43Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
heptio-authenticator-aws last commit 
```
commit 8c9b1bda7458aea47cd1917534f2222f80627c58 (HEAD -> master, origin/master, origin/HEAD)
Merge: 226f7ba c466dee
Author: Nicholas Turner <1205393+nckturner@users.noreply.github.com>
Date:   Wed May 16 14:26:40 2018 -0700

    Merge pull request #45 from rifelpet/kops_documentation
    
    Add documentation on using Authenticator with Kops
```

File: Gopkg.toml [hunk 2]
```diff
@@ -301,11 +300,15 @@ required = [
 
 [[constraint]]
   name = "k8s.io/api"
-  version = "kubernetes-1.9.1"
+  version = "kubernetes-1.10.4"
```
Comment (gyliu513) [inline Gopkg.toml]: Why api use `1.10.4` but kubernetes use `1.10`? I found that Kubernetes released `1.10.4` as well at https://github.com/kubernetes/kubernetes/releases/tag/v1.10.4

Comment (esnible) [reply Gopkg.toml]: OK, I have pushed again with 1.10.4 everywhere.

File: Gopkg.toml [hunk 3]
```diff
@@ -38,14 +38,12 @@ required = [
   name = "github.com/docker/distribution"
   branch = "master"
 
-# TODO: switch to 1.10 and newer client-go
-
-# client-go release-6.0 corresponds to k8s release-1.9 branch. See
-# https://github.com/kubernetes/client-go/tree/release-6.0#compatibility-matrix
+# client-go release-6.0 corresponds to k8s release-1.9 branch and release-7.0 is 1.10. See
```
Comment (ayj) [inline Gopkg.toml]: Can you remove the first sentence here? The compatibility-matrix reference is sufficient.

**Comment by esnible** [comment]
@ayj Thanks for approving.  I changed the comment per your request and lost your /lgtm.

File: tests/util/kube_utils.go [hunk 4]
```diff
@@ -792,14 +792,8 @@ func CreateMultiClusterSecrets(namespace string, KubeClient kubernetes.Interface
 				ServerEndpoints: nil,
 				CABundle:        nil,
 			},
-			AuthInfo: v1alpha1.AuthInfo{
-				Providers: nil,
-			},
-			CloudProvider: &v1alpha1.CloudProvider{
-				Name: "",
-			},
+			AuthInfo: v1alpha1.AuthInfo{},
```
Comment (sebastienvas) [inline tests/util/kube_utils.go]: What is this for ? If 1.9 needs what you removed put it back. We need to support the last 3 release of k8s.

Comment (esnible) [reply tests/util/kube_utils.go]: The whole reason for this PR is that K8s 1.10 _vendor/k8s.io/client-go/rest/config.go:81_ adds `ExecProvider *clientcmdapi.ExecConfig`.  Many configs need that the client-go code to use that provider.  1.10 removed these alpha version fields and the client code does not compile with them.

We should be able to talk to any K8s version that _kubectl_ can talk to with this change.

**Comment by esnible** [comment]
At @rshriram 's request I have created esnible/go1.10-k8s1.10.4-helm2.7.2-minikube0.25 on the public Docker hub and am testing using that image which contains Kubernetes 1.10.4.  This PR will be incomplete until I change _.circleci/config.yml_ to refer to the Istio version instead of my private version.

**Comment by sebastienvas** [comment]
you might want to update https://github.com/istio/istio/blob/master/prow/e2e-suite.sh#L47 to use gke-e2e-test-1-10 by default

**Pull request merged**